### PR TITLE
Add standard API endpoints for reports

### PR DESCRIPTION
**Pull request by dadarya0**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation-new/pull/180
| Issue(s) addressed                     | Fixes #... 

#### Description:
Exposure to API endpoints for reports which allows create, update and delete. We currently do not have an endpoint to get descriptions for the reports (sources available, columns available to the source, etc) so likely requires building in the UI then fetching the data to know what to use to create future reports.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Try to create a new report
```
curl -X POST \
  https://your-mautic-url.com/api/reports/new \
  -H 'content: application/json' \
  -d '{
    "name": "New Report",
    "description": "A new report",
    "system": true,
    "isScheduled": false,
    "source": "email.stats",
    "columns": [
        "es.date_sent",
        "es.date_read",
        "e.subject",
        "es.email_address",
        "e.id"
    ],
    "filters": [
        {
            "column": "e.is_published",
            "condition": "eq",
            "value": "1"
        }
    ],
    "tableOrder": [
        {
            "column": "es.date_sent",
            "direction": "ASC"
        }
    ],
    "graphs": [
        "mautic.email.graph.line.stats",
        "mautic.email.graph.pie.ignored.read.failed",
        "mautic.email.table.most.emails.read",
        "mautic.email.table.most.emails.sent",
        "mautic.email.table.most.emails.read.percent",
        "mautic.email.table.most.emails.failed"
    ],
    "groupBy": null,
    "settings": [],
    "scheduleUnit": null,
    "toAddress": null,
    "scheduleDay": null,
    "scheduleMonthFrequency": null
}
```
3. Update a report
```
curl -X PATCH \
   https://your-mautic-url.com/api/reports/7/edit \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -d name=Updated%20Report

```

4. Delete a report
```
curl -X DELETE \
  https://your-mautic-url.com/api/reports/7/delete 

```
5. Ensure that /reports and /report/ID still work as expected.

### CODE REVIEW

File: app/bundles/ReportBundle/Controller/Api/ReportApiController.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function __construct(CorePermissions $security, Translator $translator, E
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function getReportAction(Request $request, $id)
+    public function getEntityAction(Request $request, $id)
```
Comment (rohitp19) [inline app/bundles/ReportBundle/Controller/Api/ReportApiController.php:54]:
```suggestion
    public function getEntityAction(Request $request, int $id): Response
```

**Review by rohitp19 — CHANGES_REQUESTED** [review]

File: app/bundles/ReportBundle/Controller/Api/ReportApiController.php (hunk 1 above)
Comment (dadarya0) [reply app/bundles/ReportBundle/Controller/Api/ReportApiController.php:54]: We can not change signature of his method as Declaration must be compatible with [FetchCommonApiController->getEntityAction()](https://github.com/mautic/mautic/blob/dbfce0368858377035998c3c1acafa51b0815968/app/bundles/ApiBundle/Controller/FetchCommonApiController.php#L306)

Comment (rohitpavaskar) [reply app/bundles/ReportBundle/Controller/Api/ReportApiController.php:54]: Thanks for pointing it.

**Review by rohitpavaskar — APPROVED** [review]

**Comment by escopecz** [comment]
@dadarya0 could you please add the new endpoints also to [this doc](https://github.com/mautic/developer-documentation-new/blob/5.x/docs/rest_api/reports.rst) so users are aware they exist?

**Review by Galokh66 — APPROVED** [review]
Tested and verified POST/PATCH/PUT/GET/DELETE operations for /reports endpoint

**Comment by dadarya0** [comment]
> @dadarya0 could you please add the new endpoints also to [this doc](https://github.com/mautic/developer-documentation-new/blob/5.x/docs/rest_api/reports.rst) so users are aware they exist?

PR - https://github.com/mautic/developer-documentation-new/pull/180

**Pull request merged**